### ISSUE TITLE
Stair Climbing

### ISSUE THREAD

**Issue by itaouil** [graiola/wolf-setup#3]
Hi Gennaro :),

I am interested to use your work for a stair climbing approach with vision targeted to the AlienGo robot.

In the paper it is stated that the wbc might have stability issues when dealing with high inclinations, but I was wondering whether the wbc could potentially deal with stair climbing (16cm steps) or not out of the box (I am yet to try blind stair climbing in simulation as I dont have a joypad yet)?

Best,
Ilyass

**Comment by EnricoMingo** [comment]
Dear @itaouil, could you move the video to this discussion, please?

**Comment by itaouil** [comment]
Video of blind stairs climbing of aliengo in the simulator originally posted in the discussion #4: https://drive.google.com/file/d/1vKy0tUloRkoicj5D8c86OecrJgv95DCX/view?usp=sharing

**Comment by EnricoMingo** [comment]
@itaouil, did you try to address the stairs using the crawl?

**Comment by itaouil** [comment]
@EnricoMingo not yet.

I will do further tests today or tomorrow and will update you :).

**Comment by EnricoMingo** [comment]
Thanks, @itaouil.
consider that the controller may need to be further tuned (since we support multiple robots we did not fine-tune all of them). May I ask: do you have as well a hardware platform to eventually try your work?

**Comment by itaouil** [comment]
Yes.

We have an AlienGo robot here where I could try the controller out possibly, but first I want to make sure that it is robust enough in simulation.

Regarding the fine tuning process, is there any resource/package/software you would recommend to find the best parameters for the robot that could improve robustness?

**Comment by EnricoMingo** [comment]
I am sorry but I am not really aware of packages for tuning.

**Comment by graiola** [comment]
@itaouil could you send me the stairs model? I am curious to try it as well :)

**Comment by itaouil** [comment]
@graiola absolutely :)

Here is the stairs model I am using:

[stairs.zip](https://github.com/graiola/wbc-setup/files/7569069/stairs.zip)

**Comment by itaouil** [comment]
Hi @graiola and @EnricoMingo,

I tried the new debs that you pushed for issue #5 to see if the robot is now able to also climb stairs blindly.

Here is the result https://drive.google.com/file/d/1vFOuyofax-AGkbqvw2kgO0hx1W63Al7I/view?usp=sharing

**Spoiler alert** It managed :). The only thing that I noticed and that I also raised in issue #5 is the joint violations messages that  the wbc throws, which I guess is due to the "weird" joint positions that the robot acquires as soon as it starts moving, but let's keep the discussion about that in the other issue #5.

One final question, did you also include the adaptation of the base attitude based on the slope in the new debs update, or this was something already there before?

**Comment by EnricoMingo** [comment]
Dear @itaouil, this is amazing! Thank you for sharing! 
We are working on any aspect of the framework, in particular, concerning the whole-body control stack, we are trying different combinations of tasks and constraints (as well priorities). Our main goal is to try to achieve a stable walk based on reduced feedback data. The base adaptation was not present in the debians before but it was present in another branch. Since we changed the stack quite a lot we decided to put it back.
